Middleware

KilianBre opened the pull request.

arthur-crahe committed:
setting up next js

arthur-crahe committed:
Zustand & Folder architecture with README files

Created folder architecture with readme files. Installed Zustand which is a store manager for javascript applications.

arthur-crahe committed:
Update frontend README and restructured folders

Capryc0rne committed:
Initial commit backend

Capryc0rne committed:
updated README.md

Capryc0rne committed:
remove .idea directory

Capryc0rne committed:
updated gitignore
- exclude .idea/

arthur-crahe committed:
installed & configured antd

arthur-crahe committed:
Installed Axios ; Removed CSS

Capryc0rne committed:
init Database

Capryc0rne committed:
Merge remote-tracking branch 'origin/setting_up_project' into setting_up_project

Capryc0rne committed:
init authentication
+ Dumb DB test

arthur-crahe committed:
init front authentication

accessible at localhost/login

Capryc0rne committed:
CORS Origin fix
+ → http://localhost:3000

arthur-crahe committed:
disable signup ; regex for email ; handling error codes ;

Capryc0rne committed:
Refine userData return on login

arthur-crahe committed:
Regex signup ; Logout component

Displaying logout component or login/signup depending on the "user" value.
Inscreased timeout in Axios requests because we had some timeout issues.

arthur-crahe committed:
Merge branch 'authentication' of https://github.com/CESI-GRP-4/ressources_relationnelles into authentication

Capryc0rne committed:
Signup backend module
+ /api/signup

arthur-crahe committed:
Fixed logout ; Fixed login response ; Regex ; Error handling

Fixed logout : remove cookie when user is null

Fixed login and signup response : we expected data to be directly accessible but user's data is actually in the "user" object of the response.

More maintainable regex expression : they are now exportable for more consistency.

Error handling : [sign up] handling codes when falling in the catch statement.

Capryc0rne committed:
Merge branch 'dev' into authentication

KilianBre committed:
Create middleware.ts

KilianBre committed:
Test cypress

arthur-crahe reviewed (changes requested):
React hooks are not accessible from the Middleware. Instead of using useUser, you must call an api which will return the validity of the JWT (see with @Capryc0rne or @Tristan13R).

PS: syntax

`const { user, setUser }  = useUser(); `

PS: Why is there so many files to update ???? Make your PR into [Authentication](https://github.com/CESI-GRP-4/ressources_relationnelles/tree/authentication) and not Dev.

arthur-crahe committed:
Moved to subfolders ; Renamed routes ; Layouts ; Forgot password link

- Moved pages in corresponding subfolder front-office or back-office.
- Renamed routes in french.
- Created layout for each page in order to change metadata.
- Created a "forgot password" link component

Capryc0rne committed:
Add Email Verification Feature
+ Update Database Schema

Capryc0rne committed:
Merge remote-tracking branch 'origin/authentication' into authentication

Capryc0rne committed:
Add Password Recovery Feature
+ Update Database Schema : ressources_relationnelles.sql

Capryc0rne committed:
Add Remember me Feature
+ set TTL JWT at 14 days

KilianBre committed:
ok

arthur-crahe committed:
Reset password ; Mail verification ; Metadata ; Password input

arthur-crahe committed:
Merge branch 'authentication' of https://github.com/CESI-GRP-4/ressources_relationnelles into authentication

arthur-crahe committed:
new login/sign up response type.

Capryc0rne committed:
Add NewUser Feature

Capryc0rne committed:
Merge remote-tracking branch 'origin/authentication' into authentication

Capryc0rne committed:
Init Swagger: API
+ api-docs.json -> editor.swagger.io

+ composer install

Capryc0rne committed:
JWT auto Refresh: API

Capryc0rne committed:
API update + add path /api/verifyUser

+ pass "newUser" in "user" response

arthur-crahe committed:
Merge branch 'dev' into authentication

KilianBre committed:
README cypress

KilianBre committed:
Update README.md

KilianBre committed:
Update loginPage.cy.ts

KilianBre committed:
Update (newUser dans user)

KilianBre committed:
Merge branch 'authentication' of https://github.com/CESI-GRP-4/ressources_relationnelles into auth/kilian

KilianBre committed:
middleware update
